internal/resources/providers/gcplib/inventory/provider.go

```diff
@@ -91,460 +75,397 @@ type dnsPolicyFields struct {
 	enableLogging bool
 }
 
-type TypeGenerator[T any] func(assets []*ExtendedGcpAsset, projectId, projectName, orgId, orgName string) *T
-
 type Iterator interface {
 	Next() (*assetpb.Asset, error)
 }
 
 type ServiceAPI interface {
-	// ListAllAssetTypesByName List all content types of the given assets types
-	ListAllAssetTypesByName(ctx context.Context, assets []string) ([]*ExtendedGcpAsset, error)
-
-	// ListMonitoringAssets List all monitoring assets by project id
-	ListMonitoringAssets(ctx context.Context, monitoringAssetTypes map[string][]string) ([]*MonitoringAsset, error)
-
-	// ListLoggingAssets returns a list of logging assets grouped by project id, extended with folder and org level log sinks
-	ListLoggingAssets(ctx context.Context) ([]*LoggingAsset, error)
-
-	// ListServiceUsageAssets returns a list of service usage assets grouped by project id
-	ListServiceUsageAssets(ctx context.Context) ([]*ServiceUsageAsset, error)
-
-	// returns a project policies for all its ancestors
-	ListProjectsAncestorsPolicies(ctx context.Context) ([]*ProjectPoliciesAsset, error)
-
-	// Close the GCP asset client
+	ListAssetTypes(ctx context.Context, assetTypes []string, out chan<- *ExtendedGcpAsset)
+	ListMonitoringAssets(ctx context.Context, out chan<- *MonitoringAsset)
+	ListProjectsAncestorsPolicies(ctx context.Context, out chan<- *ProjectPoliciesAsset)
+	ListProjectAssets(ctx context.Context, assetTypes []string, out chan<- *ProjectAssets)
+	ListNetworkAssets(ctx context.Context, out chan<- *ExtendedGcpAsset)
+	Clear()
 	Close() error
 }
 
 type ProviderInitializerAPI interface {
-	// Init initializes the GCP asset client
 	Init(ctx context.Context, log *clog.Logger, gcpConfig auth.GcpFactoryConfig) (ServiceAPI, error)
 }
 
-func (p *ProviderInitializer) Init(ctx context.Context, log *clog.Logger, gcpConfig auth.GcpFactoryConfig) (ServiceAPI, error) {
+func newAssetsInventoryWrapper(ctx context.Context, log *clog.Logger, gcpConfig auth.GcpFactoryConfig) (*AssetsInventoryWrapper, error) {
 	limiter := NewAssetsInventoryRateLimiter(log)
```

Comment:
we use rate limiting for the GCP Assets Inventory `ListAssets` method, and because we're making more concurrent calls, the cycle fetching takes longer (~30seconds vs ~18secs) as we wait between calls.